hieradata/clients/cert-ci.yaml

```diff
@@ -1,44 +1,10 @@
 ---
 profile::buildmaster::ci_fqdn: 'cert.ci.jenkins.io'
+profile::buildmaster::groovy_d_agent_security: 'absent'
```

Comment:
This looks like opt-out, which is likely error prone…?

Reply:
Puppet resources use the parameter ensure => "present or absent" to know if a resource must be installed or removed. In this case it's used [here](https://github.com/jenkins-infra/jenkins-infra/blob/72adc28ca884e47ae15f8ad9302ba8268b8b0580/dist/profile/manifests/buildmaster.pp#L145) and means that for the cert-ci instance, that groovy script must be absent

Reply:
When someone adds a new Groovy script for use by ci.jenkins.io, do they need to remember to edit this file? If so, that seems like a problem as that would easily be missed.

Reply:
This puppet class only install/desinstall groovy script defined in this puppet repository. 
The approach is:
1) If you don't need templating, you can add your specific groovy script in this [directory](https://github.com/jenkins-infra/jenkins-infra/tree/b461b54dbb50101a9ee2552e37219775df278916/dist/profile/files/buildmaster)
2) If you need to templating, put your template in this [directory](https://github.com/jenkins-infra/jenkins-infra/tree/b461b54dbb50101a9ee2552e37219775df278916/dist/profile/files/buildmaster) like [lockbox.groovy.erb](https://github.com/jenkins-infra/jenkins-infra/blob/staging/dist/profile/templates/buildmaster/lockbox.groovy.erb)
 3) Update the buildmaster puppet [manifest](https://github.com/jenkins-infra/jenkins-infra/blob/b461b54dbb50101a9ee2552e37219775df278916/dist/profile/manifests/buildmaster.pp)
with your groovy script either based on a [emplate](https://github.com/jenkins-infra/jenkins-infra/blob/b461b54dbb50101a9ee2552e37219775df278916/dist/profile/manifests/buildmaster.pp#L188-L178) or based on a [file](https://github.com/jenkins-infra/jenkins-infra/blob/b461b54dbb50101a9ee2552e37219775df278916/dist/profile/manifests/buildmaster.pp#L176-L166)

Next time puppet agent run it will read and apply the manifest.

Parameter are customized per node from this [directory](https://github.com/jenkins-infra/jenkins-infra/tree/b461b54dbb50101a9ee2552e37219775df278916/hieradata/clients)